spec/facets/ProcessingTypeJobFacet.json

```diff
@@ -0,0 +1,31 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "https://openlineage.io/spec/facets/2-0-2/ProcessingTypeJobFacet.json",
+  "$defs": {
+    "ProcessingTypeJobFacet": {
```

Comment:
I think, this is an opportunity to more closely define job types like we do Naming for Datasets.
I would propose to do the following:
```
jobType: {
  processingType: enum: "STREAM|BATCH"
  integration: STRING  ex: "SPARK|DBT|AIRFLOW|FLINK"
  jobType: STRING ex:"QUERY|COMMAND|DAG|TASK|JOB|MODEL|..."
}
```
Then the naming documentation for jobs would define:
- integration: SPARK
   - processingType: "STREAM|BATCH"
   - jobType: "JOB|COMMAND"
- integration: AIRFLOW
   - processingType: "BATCH"
   - jobType: "DAG|TASK"
- integration: DBT
   - processingType: "BATCH"
   - jobType: "PROJECT|MODEL"
- integration: FLINK
   - processingType: "STREAM|BATCH"
   - jobType: "JOB"

This doc would also clarify usage of parent jobs.
This will enable better visualization of jobs as well.

Reply:
Docs included in https://github.com/OpenLineage/docs/pull/257

Reply:
This duplicates `ProcessingEngineRunFacet`, but I like it. Generally I think this makes much more sense as job-level facet rather than run-based one, since the processing engine property wouldn't really change between job runs.

I think a very nice property of that facet is being able to distinguish whether the event is a "wrapper" job or a "leaf" job - something that we had to do "bottom-up" - only knowledge we had about it was either by treating job name in a semantically, or by looking at "children job" references. This solves the problem by making it explicite.

Reply:
Yes, I was also initially afraid of duplicating `ProcessingEngineRunFacet`. My current understanding is that it makes sense to have both. If airflow triggers snowflake SQL, then integration is `airflow` while processing engine should be `snowflake`, so integration has slightly different meaning: it tells more who is sending the event than what is processing data.

Reply:
I like an idea of having both of the facets and see need for them.